polyfactory/factories/pydantic_factory.py

```diff
@@ -164,3 +171,50 @@ def build(cls, factory_use_construct: bool = False, **kwargs: Any) -> T:
             return cls.__model__.construct(**processed_kwargs)
 
         return cls.__model__(**processed_kwargs)
+
+    @classmethod
+    def get_provider_map(cls) -> dict[Any, Callable[[], Any]]:
```

Comment:
this change is not desirable. The reason for setting the pydantic types on the base level is that pydantic allows using its special types in dataclasses and typeddict as well. It thus cannot be isolated to this factory.

Reply:
I see, thanks; reverted this commit.